lib/kafka_ex/consumer_group.ex

```diff
@@ -135,14 +135,33 @@ defmodule KafkaEx.ConsumerGroup do
 
   This function has the same return values as `Supervisor.start_link/3`.
   """
-  @spec start_link(module, binary, [binary], options) :: Supervisor.on_start()
-  def start_link(consumer_module, group_name, topics, opts \\ []) do
+  @spec start_link(module | {module, module}, binary, [binary], options) :: Supervisor.on_start()
+  def start_link(consumer_module, group_name, topics, opts \\ [])
+  def start_link(consumer_module, group_name, topics, opts)
+      when is_atom(consumer_module) do
+    start_link({KafkaEx.GenConsumer, consumer_module}, group_name, topics, opts)
+  end
+
+  @doc """
```

Comment:
I think this `@doc` needs to be combined with the other function head's `@doc` for the documentation to be generated correctly?